pkg/noderesourcetopology/cache/overallocating.go

```diff
@@ -0,0 +1,221 @@
+
+	nodeAssumedResources.AddPod(pod)
+	klog.V(5).InfoS("nrtcache post reserve", "logKey", logKey, "node", nodeName, "assumedResources", nodeAssumedResources.String())
+
+	ov.nodeDiscarded.Delete(nodeName)
+	klog.V(6).InfoS("nrtcache: reset discard counter", "logKey", logKey, "node", nodeName)
+}
+
+func (ov *Overallocating) ReleaseNodeResources(nodeName string, pod *corev1.Pod) {
+	logKey := keyFromPod(pod)
+	ov.lock.Lock()
+	defer ov.lock.Unlock()
+	nodeAssumedResources, ok := ov.assumedResources[nodeName]
+	if !ok {
+		// this should not happen, so we're vocal about it
+		// we don't return error because not much to do to recover anyway
+		klog.V(3).InfoS("nrtcache: no resources tracked", "logKey", logKey, "node", nodeName)
+		return
+	}
+
+	nodeAssumedResources.DeletePod(pod)
+	klog.V(5).InfoS("nrtcache post release", "logKey", logKey, "node", nodeName, "assumedResources", nodeAssumedResources.String())
+}
+
+// DirtyNodeNames returns a slice of all the node names which have been discarded previously,
+// so which are supposed to be `dirty` in the cache.
+// A node can be discarded for two reasons:
+// 1. it legititmately cannot fit containers because it has not enough free resources
+// 2. it was pessimistically overallocated, so the node is a candidate for resync
+// This function enables the caller to know the slice of nodes should be considered for resync,
+// avoiding the need to rescan the full node list.
+func (ov *Overallocating) DirtyNodeNames(logKey string) []string {
+	ov.lock.Lock()
+	defer ov.lock.Unlock()
+	var nodes []string
+	for node := range ov.assumedResources {
+		if !ov.nodeDiscarded.IsSet(node) {
+			// noone asked about this node, let it be
+			continue
+		}
+		nodes = append(nodes, node)
+	}
+
+	if len(nodes) > 0 {
+		klog.V(4).InfoS("nrtcache: found dirty nodes", "logKey", logKey, "nodes", len(nodes))
+	}
+	return nodes
+}
+
+// Resync implements the cache resync loop step. This function checks if the latest available NRT information received matches the
+// state of a dirty node, for all the dirty nodes. If this is the case, the cache of a node can be Flush()ed.
+func (ov *Overallocating) Resync() {
+	// we are not working with a specific pod, so we need a unique key to track this flow
+	logKey := shortuuid.New()
+
+	klog.V(6).InfoS("nrtcache: resync NodeTopology cache starting", "logKey", logKey)
+	defer klog.V(6).InfoS("nrtcache: resync NodeTopology cache complete", "logKey", logKey)
+	nodeNames := ov.DirtyNodeNames(logKey)
+
+	var nrtUpdates []*topologyv1alpha1.NodeResourceTopology
+	for _, nodeName := range nodeNames {
+		nrtCandidate, err := ov.lister.Get(nodeName)
+		if err != nil || nrtCandidate == nil {
+			klog.V(3).InfoS("nrtcache: missing NodeTopology", "logKey", logKey, "node", nodeName)
+			continue
+		}
+
+		pfpExpected := podFingerprintForNodeTopology(nrtCandidate)
+		if pfpExpected == "" {
+			klog.V(3).InfoS("nrtcache: missing NodeTopology podset fingerprint data", "logKey", logKey, "node", nodeName)
```

Comment:
that's true. I'm not sure how can we handle in a more explicit way though. Missing the fingerprint from the node cause that node to be considered full prematurely (because pessimistic allocation) but doesn't impact others.